LBH.AdultSocialCare.Api/V1/Gateways/HomeCarePackageGateway.cs

```diff
@@ -29,24 +30,26 @@ public async Task<HomeCarePackage> GetAsync(Guid homeCarePackageId)
 
         public async Task<IList<HomeCarePackage>> ListAsync()
         {
-            return await _databaseContext.HomeCarePackage.ToListAsync().ConfigureAwait(false);
+            return await _databaseContext.HomeCarePackage
+                .Include(item => item.Clients)
+                .Include(item => item.Status)
+                .ToListAsync().ConfigureAwait(false);
         }
 
-        public async Task<HomeCarePackage> ChangeStatusAsync(HomeCarePackage homeCarePackage)
+        public async Task<HomeCarePackage> ChangeStatusAsync(Guid homeCarePackageId, int statusId)
         {
             HomeCarePackage homeCarePackageToUpdate = await _databaseContext.HomeCarePackage
-                .FirstOrDefaultAsync(item => item.Id == homeCarePackage.Id)
+                .Include(item => item.Clients)
+                .Include(item => item.Status)
+                .FirstOrDefaultAsync(item => item.Id == homeCarePackageId)
                 .ConfigureAwait(false);
 
-            if (homeCarePackageToUpdate != null)
+            if (homeCarePackageToUpdate == null)
             {
-                homeCarePackageToUpdate.StatusId = homeCarePackage.StatusId;
-                homeCarePackageToUpdate.CreatorId = homeCarePackage.CreatorId;
-                homeCarePackageToUpdate.UpdatorId = homeCarePackage.UpdatorId;
+                throw new ErrorException($"Couldn't find the record: {homeCarePackageId}");
             }
-
+            homeCarePackageToUpdate.StatusId = statusId;
             await _databaseContext.SaveChangesAsync().ConfigureAwait(false);
-
             return homeCarePackageToUpdate;
```

Comment:
Shouldn't gateway return `[Object].ToDomain()`?

Reply:
I think gateways returns entity @petemade ?

Reply:
Gateway - entity 
UseCase - Domain
Controller - Boundary object 
@petemade ?

Reply:
@FurkanKayar55 @petemade  Before we start unit testing we need to have this sorted

Reply:
Agree